nvflare/private/fed/server/job_runner.py

```diff
@@ -328,28 +330,32 @@ def _save_workspace(self, fl_ctx: FLContext):
         shutil.rmtree(run_dir)
 
     def run(self, fl_ctx: FLContext):
+        """Starts job runner."""
         engine = fl_ctx.get_engine()
-
-        thread = threading.Thread(target=self._job_complete_process, args=[fl_ctx])
-        thread.start()
-
         job_manager = engine.get_component(SystemComponents.JOB_MANAGER)
         if job_manager:
+            thread = threading.Thread(target=self._job_complete_process, args=[fl_ctx])
+            thread.start()
+
             while not self.ask_to_stop:
-                # approved_jobs = job_manager.get_jobs_by_status(RunStatus.APPROVED, fl_ctx)
+                if not isinstance(engine.server.server_state, HotState):
```

Comment:
Now you changed to put the check into the loop - good.

You could put the check as an "if" condition for rest of the logic to reuse the existing sleep(1.0). This is just coding style thing.